File: Tarun/Array/Sort Colors.java
class Solution {
    public void swap(int[] nums, int a, int b)
    {
        int temp=nums[a];
        nums[a]=nums[b];
        nums[b]=temp;
    }
    public void sortColors(int[] nums) {
        int start = 0;
        int end = nums.length-1;
        int index = 0;
        while(index <= end)
        {
            if(nums[index]==0)
            {
                swap(nums,start,index);
                index++;
                start++;
            }
            else if(nums[index]==2)
            {
                swap(nums,end,index);
                end--;
            }
            else
            index++;
        }
    }
}

File: Divyansh Singh/Array/sort colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
      // using Dutch Flag Algorithm
        int i=0,start=0,end=nums.size()-1;
        while(i<=end){
            if(nums[i]==2){
                swap(nums[i],nums[end]);
                end--;
            }
            else if(nums[i]==0){
                swap(nums[i],nums[start]);
                i++,start++;
            }
            else{
                i++;
            }
        }
    }
};
